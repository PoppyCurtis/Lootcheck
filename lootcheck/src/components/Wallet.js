import React, {Component} from 'react';
import { connect } from 'react-redux';
import balance from '../reducers/balance';


export class Wallet extends Component {
    render () {
        return (
            <div>
                <h3 className='balance'>Wallet balance: {this.props.balance}</h3>
            </div>
        );
    }
}
// connect function takes 2 params. 
//param 1 = what part of Redux store we want
//param 2 = what action creators we want to use on the component to send data to redux store
//this maps state to props
export default connect(state => {balance: state} , null)(Wallet);