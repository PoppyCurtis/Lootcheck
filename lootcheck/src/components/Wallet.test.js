import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Wallet', () => {
    const props = { balance: 20 }
    const wallet = shallow(<Wallet {...props}/>);
    it('renders properly', () => {
        expect(wallet).toMatchSnapshot();
    });

    it('displays the balance from props', () => {
        expect(wallet.find('.balance').text()).toEqual('Wallet balance: 20');
    });
});