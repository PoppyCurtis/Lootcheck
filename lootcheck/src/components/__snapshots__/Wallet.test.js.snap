// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `ShallowWrapper {}`;
